cupy/cuda/nccl.pyx

```diff
@@ -113,22 +142,44 @@ cdef class NcclCommunicator:
         return rank_id
 
     def allReduce(self, size_t sendbuf, size_t recvbuf,
-                  int count, int datatype, int op, size_t stream):
-        status = ncclAllReduce(<void*>sendbuf, <void*>recvbuf, count,
-                               <ncclDataType_t>datatype, <ncclRedOp_t>op,
-                               self._comm, <driver.Stream>stream)
+                  size_t count, int datatype, int op, size_t stream):
+        with nogil:
+            status = _ncclAllReduce(<void*>sendbuf, <void*>recvbuf,
+                                    count, <ncclDataType_t>datatype,
+                                    <ncclRedOp_t>op, self._comm,
+                                    <driver.Stream>stream)
         check_status(status)
 
     def reduce(self, size_t sendbuf, size_t recvbuf,
-               int count, int datatype, int op, int root, size_t stream):
-        status = ncclReduce(<void*> sendbuf, <void*> recvbuf, count,
-                            <ncclDataType_t> datatype, <ncclRedOp_t> op, root,
-                            self._comm, <driver.Stream> stream)
+               size_t count, int datatype, int op, int root, size_t stream):
+        with nogil:
+            status = _ncclReduce(<void*>sendbuf, <void*>recvbuf,
+                                 count, <ncclDataType_t>datatype,
+                                 <ncclRedOp_t>op, root, self._comm,
+                                 <driver.Stream>stream)
         check_status(status)
 
     def bcast(self, size_t buff, int count, int datatype,
               int root, size_t stream):
-        status = ncclBcast(<void*> buff, count,
-                           <ncclDataType_t> datatype, root,
-                           self._comm, <driver.Stream> stream)
+        with nogil:
+            status = _ncclBcast(<void*>buff, count,
+                                <ncclDataType_t>datatype, root,
+                                self._comm, <driver.Stream>stream)
+        check_status(status)
+
+    def reduceScatter(self, size_t sendbuf, size_t recvbuf,
+                      size_t recvcount, int datatype, int op, size_t stream):
+        with nogil:
+            status = _ncclReduceScatter(<void*>sendbuf, <void*>recvbuf,
+                                        recvcount, <ncclDataType_t>datatype,
+                                        <ncclRedOp_t>op, self._comm,
+                                        <driver.Stream>stream)
+        check_status(status)
+
+    def allGather(self, size_t sendbuf, size_t count, int datatype,
+                  size_t recvbuf, size_t stream):
```

Comment:
Thank you for pointing out that!

I was bit careless. I intended I was following nccl2's order of arguments there. Will fix soon.